src/language_features/document_symbol.rs

```diff
@@ -23,6 +23,125 @@
+    let eligible_servers: Vec<_> = ctx
+        .language_servers
+        .iter()
+        .filter(|srv| attempt_server_capability(*srv, &meta, CAPABILITY_DOCUMENT_SYMBOL))
+        .collect();
+    let req_params = eligible_servers
+        .into_iter()
+        .map(|(server_name, _)| {
+            (
+                server_name.clone(),
+                vec![DocumentSymbolParams {
+                    text_document: TextDocumentIdentifier {
+                        uri: Url::from_file_path(&meta.buffile).unwrap(),
+                    },
+                    partial_result_params: Default::default(),
+                    work_done_progress_params: Default::default(),
+                }],
+            )
+        })
+        .collect();
+    let params = BreadcrumbsParams::deserialize(editor_params).unwrap();
+    ctx.call::<DocumentSymbolRequest, _>(
+        meta,
+        RequestParams::Each(req_params),
+        move |ctx: &mut Context, meta, results| {
+            match results.into_iter().find(|(_, v)| v.is_some()) {
+                Some((server_name, Some(DocumentSymbolResponse::Nested(symbols))))
+                    if !symbols.is_empty() =>
+                {
+                    let server = &ctx.language_servers[&server_name];
+                    let document = if let Some(document) = ctx.documents.get(&meta.buffile) {
+                        document
+                    } else {
+                        return;
+                    };
+                    let mut filename_path = PathBuf::default();
+                    let symbol = &symbols[0];
+                    let filename = symbol_filename(&meta, symbol, &mut filename_path).to_string();
+                    let mut breadcrumbs = Vec::default();
+                    breadcrumbs_calc(
+                        &symbols,
+                        &params,
+                        ctx,
+                        server,
+                        document,
+                        filename.as_str(),
+                        &mut breadcrumbs,
+                    );
+                    if let Some(last) = breadcrumbs.last_mut() {
+                        *last = format!("{last} ");
+                    };
+                    let mut breadcrumbs = breadcrumbs.join(" > ");
+                    breadcrumbs = editor_quote(breadcrumbs.as_str());
+                    let command =
+                        format!("set-option buffer lsp_modeline_breadcrumbs {breadcrumbs}");
+                    let command = format!(
+                        "evaluate-commands -buffer {} -- {}",
+                        editor_quote(&meta.buffile),
+                        editor_quote(&command)
+                    );
+                    ctx.exec(meta, command);
+                }
+                Some((_, Some(DocumentSymbolResponse::Nested(_)))) => {
+                    return;
+                }
+                Some((_, Some(DocumentSymbolResponse::Flat(_)))) => {
```

Comment:
It will be very easy to support Flat symbol responses.
We have a `trait Symbol<T: Symbol<T>>` that is an umbrella for both flat and nested symbols
that can absolutely be done post-merge but I did it already